### PR TITLE
insights: remove deprecated metadata mentions

### LINKED ISSUE #37719: insights: deprecate mentions of `metadata`
metadata was added as something we thought we might need, but we don't actually use it. 

- [x] * remove `metadata` fields in structs
- [x] * remove db calls

 remove the db table to be done in separate ticket

### PR DESCRIPTION
**Pull request by leonore**
closes #37719

## Test plan

Ran existing unit tests.

### CODE REVIEW

File: migrations/codeinsights/1663782457_remove_metadata_table/up.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE IF EXISTS series_points DROP COLUMN IF EXISTS metadata_id;
```
Comment (coury-clark) [inline migrations/codeinsights/1663782457_remove_metadata_table/up.sql:1]: Thank you SO much for taking the time to deprecate this!

One thing though, we have a backwards compatibility requirement, we will need to mutate the DB 1 version after we move the code references here.

**Review by coury-clark — CHANGES_REQUESTED** [review]
Left a note below! Super happy to see this, but we will need to be migration compatible here! Let me know if there is any more information I can provide

File: migrations/codeinsights/1663782457_remove_metadata_table/up.sql (hunk 1 above)
Comment (leonore) [reply migrations/codeinsights/1663782457_remove_metadata_table/up.sql:1]: you know I love a good deprecation 😄  I think you mean that the metadata table needs to be removed in 4.2 (and the code in 4.1). I've created #41937 for that, and will remove the changes here for now!

Comment (coury-clark) [reply migrations/codeinsights/1663782457_remove_metadata_table/up.sql:1]: Right, and the DB references. In other words, 4.0 needs to be able to run against this schema

Comment (coury-clark) [reply migrations/codeinsights/1663782457_remove_metadata_table/up.sql]: Sorry, that wasn't very clear. I mean to say you can freely change anything in the app code, but don't touch the schema until +1 versions out.

**Pull request merged**